os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -275,65 +275,74 @@ static void up_dumpstate(void)
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
 	lldbg("sp:     %08x\n", sp);
 
-#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
-	/* Get the limits on the nested irq stack */
-	nestirqstkbase = (uint32_t)&g_nestedirqstkbase;
-	nestirqstksize = (CONFIG_ARCH_NESTED_IRQ_STACK_SIZE & ~3);
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler
+	 */
 
-	lldbg("Nested IRQ stack:\n");
-	lldbg("  base: %08x\n", nestirqstkbase);
-	lldbg("  size: %08x\n", nestirqstksize);
+	if (current_regs) {
+#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
+		lldbg("Code asserted in nested IRQ state!\n");
+#else
+		lldbg("Code asserted in IRQ state!\n");
+#endif
+		lldbg("IRQ num: %d \n", getipsr());
+		lldbg("IRQ handler: %08x \n", g_irqvector[getipsr()].handler);
+#ifdef CONFIG_DEBUG_IRQ_INFO
+		lldbg("IRQ handler: %s \n", g_irqvector[getipsr()].irq_name);
+#endif
```

Comment:
This change will cause a build error because the if and else are not matching based on the configs being enabled. 
Also, the irq stack information is not related to whether the crash happened in irq context or not. So, pls change the implementation as follows:

```
if (crash in irq)
    print irq related info
else 
    print that crash is in process context
end

Keep remaining logic for printing the stack as before
...
...
```

Reply:
Thank you for pointing this out Kishore. I have modified the logic to fix the if else discrepency.
But we had concluded in the chat that we do not need to print interrupt stack until and unless crash is in irq state...So I'm not sure if we should put the "Keep remaining logic for printing the stack as before" outside the *if (crash in irq)* loop

Reply:
Ok.. If the crash has happened in user code, then the irq stack will not contain any valuable information. So, it is not required to print it.